app.rb

```diff
@@ -0,0 +1,180 @@
+    end
+  end
+  # rubocop:enable Metrics/CyclomaticComplexity
+
+  def list_books
+    puts '📕📗📘📙'
+    @books.each do |book|
+      puts "#{book.title} by #{book.author}"
+    end
+    puts ''
+    run
+  end
+
+  def list_people
+    @people.each do |person|
+      puts "[#{person.class.name}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+    puts ''
+    run
+  end
+
+  def create_person
+    puts 'Do you want to create a student (1) or a teacher (2)?'
+    input = gets.chomp.to_i
+
+    case input
+    when 1
+      create_student
+    when 2
+      create_teacher
+    end
+
+    run
+  end
+
+  def create_student
+    puts 'Age: '
+    age = gets.chomp.to_i
+    puts 'Name: '
+    name = gets.chomp
+
+    puts 'Has parent permission? [Y/N]'
+    permission = gets.chomp.upcase
+    case permission
+    when 'Y'
+      permission = true
+    when 'N'
+      permission = false
+    end
+
+    student = Student.new(nil, age, name, parent_permission: permission)
+    @people.push(student)
+    puts "Student (#{student.name}) created successfully"
+    puts ''
+  end
+
+  def create_teacher
+    puts 'Age: '
+    age = gets.chomp.to_i
+    puts 'Specialization: '
+    specialization = gets.chomp
+    puts 'Name: '
+    name = gets.chomp
+
+    puts 'Has parent permission? [Y/N]'
+    permission = gets.chomp.upcase
+    case permission
+    when 'Y'
+      permission = true
+    when 'N'
+      permission = false
+    end
+
+    teacher = Teacher.new(age, specialization, name, parent_permission: permission)
+    @people.push(teacher)
+    puts "Teacher (#{teacher.name}) created successfully"
+    puts ''
+  end
+
+  def create_book
+    puts 'Title:'
+    title = gets.chomp
+    puts 'Author:'
+    author = gets.chomp
+    book = Book.new(title, author)
+    @books.push(book)
+    puts "Book #{book.title} created successfully ✔️"
+    puts ''
+    run
+  end
+
+  def create_rental
+    puts 'Select the book from the following list by number (not ID)'
+    @books.each_with_index do |book, index|
+      puts "#{index}) Title: '#{book.title}', Author: '#{book.author}'"
+    end
+    selected_book = gets.chomp.to_i
+    puts 'Date:'
+    date = gets.chomp
+
+    puts 'Select a person from the following list by number (not ID)'
+    @people.each_with_index do |person, index|
+      puts "#{index}) [#{person.class.name}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+    selected_person = gets.chomp.to_i
+
+    rental = Rental.new(date, @books[selected_book], @people[selected_person])
+    @rentals.push(rental)
+    puts 'Rental created successfully'
+    puts ''
+    run
+  end
+
+  def list_rental
+    puts 'ID of person'
+    id = gets.chomp.to_i
+    puts "Rentals by PersonID:#{id}"
+    rentals = @rentals.select { |rental| id == rental.person.id }
+    rentals.each do |item|
+      puts "Date: #{item.date},  Book #{item.book.title}, by #{item.book.author}"
+      puts ''
+      run
+    end
+  end
```

Comment:
- [ ] When I try to list rental for a person, if I select the ID of a person that does not have a rental the app just quits. From the video demo this scenario should be handled as well. You can just show an empty result, like the video, and still maintain the running app. Kindly fix that.